src/runtime/task/mod.rs

```diff
@@ -197,7 +197,6 @@ impl Task {
     /// waiter should block or not. If false, this task has already finished, and so the waiter need
     /// not block.
     pub(crate) fn set_waiter(&mut self, waiter: TaskId) -> bool {
-        assert!(self.waiter.is_none());
```

Comment:
Rather than removing this assertion entirely, can we weaken it to:
```
assert!(self.waiter.is_none() || self.waiter == Some(waiter));
```
The point being that Shuttle can't currently handle multiple tasks waiting on the same other task. This way we'll still detect that situation happening rather than just silently missing behaviors.

Reply:
Ah I didn't realize that. Maybe we can add a message to this assert like "Task cannot have more than one waiter" to make this easier to debug?

Reply:
yep, that message looks good to me